app/src/main/java/org/schabi/newpipe/subscription/services/BaseImportExportService.java

```diff
@@ -0,0 +1,203 @@
+
+import org.schabi.newpipe.R;
+import org.schabi.newpipe.extractor.subscription.SubscriptionExtractor;
+import org.schabi.newpipe.report.ErrorActivity;
+import org.schabi.newpipe.report.UserAction;
+import org.schabi.newpipe.subscription.ImportExportEventListener;
+import org.schabi.newpipe.subscription.SubscriptionService;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.disposables.CompositeDisposable;
+import io.reactivex.processors.PublishProcessor;
+
+public abstract class BaseImportExportService extends Service {
+    protected final String TAG = this.getClass().getSimpleName();
+
+    protected NotificationManagerCompat notificationManager;
+    protected NotificationCompat.Builder notificationBuilder;
+
+    protected SubscriptionService subscriptionService;
+    protected CompositeDisposable disposables = new CompositeDisposable();
+    protected PublishProcessor<String> notificationUpdater = PublishProcessor.create();
+
+    @Nullable
+    @Override
+    public IBinder onBind(Intent intent) {
+        return null;
+    }
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+        subscriptionService = SubscriptionService.getInstance(this);
+        setupNotification();
+    }
+
+    @Override
+    public void onDestroy() {
+        super.onDestroy();
+        disposeAll();
+    }
+
+    protected void disposeAll() {
+        disposables.clear();
+    }
+
+    /*//////////////////////////////////////////////////////////////////////////
+    // Notification Impl
+    //////////////////////////////////////////////////////////////////////////*/
+
+    private static final int NOTIFICATION_SAMPLING_PERIOD = 2500;
+
+    protected AtomicInteger currentProgress = new AtomicInteger(-1);
+    protected AtomicInteger maxProgress = new AtomicInteger(-1);
+    protected ImportExportEventListener eventListener = new ImportExportEventListener() {
+        @Override
+        public void onSizeReceived(int size) {
+            maxProgress.set(size);
+            currentProgress.set(0);
+        }
+
+        @Override
+        public void onItemCompleted(String itemName) {
+            currentProgress.incrementAndGet();
+            notificationUpdater.onNext(itemName);
+        }
+    };
+
+    protected abstract int getNotificationId();
+    @StringRes
+    public abstract int getTitle();
+
+    protected void setupNotification() {
+        notificationManager = NotificationManagerCompat.from(this);
+        notificationBuilder = createNotification();
+        startForeground(getNotificationId(), notificationBuilder.build());
+
+        disposables.add(notificationUpdater
+                .filter(s -> !s.isEmpty())
+                .publish(stringFlowable -> stringFlowable.limit(1)
```

Comment:
Took some eye squinting to get what this is doing. Might be a good idea to extract this and name it `notifyFirstThenThrottle` or something more descriptive.

Reply:
I've chosen `throttleAfterFirstEmission`, I hope that it's easier to understand.